addons/event_sale/tests/test_event_sale.py

```diff
@@ -482,6 +482,34 @@ def test_ticket_price_with_pricelist_and_tax(self):
         })
         self.assertEqual(so.amount_total, 660.0, "Ticket is $1000 but the event product is on a pricelist 10 -> 6. So, $600 + a 10% tax.")
 
+    def test_event_sale_price_subtotal(self):
+        """Check that the `sale_price_subtotal` should be the total price of only the confirmed orders."""
+        ticket, ticket1 = self.event_0.event_ticket_ids
+        vals_list = [
+            {'ticket': ticket1, 'price': 195.50, 'state': 'sale'},
+            {'ticket': ticket, 'price': 200.50, 'state': 'sent'},
+            {'ticket': ticket1, 'price': 205.50, 'state': 'cancel'},
+        ]
+        self.env['sale.order'].create([{
+            'partner_id': self.env.user.partner_id.id,
+            'state': vals['state'],
+            'order_line': [
+                (0, 0, {
+                    'event_id': self.event_0.id,
+                    'event_ticket_id': vals['ticket'].id,
+                    'product_id': vals['ticket'].product_id.id,
+                    'price_unit': vals['price'],
+                })
+            ]
+        } for vals in vals_list])
+
+        for order_line, expected_state in zip(self.event_0.sale_order_lines_ids, ['cancel', 'sent', 'sale', 'draft']):
+            self.assertEqual(order_line.state, expected_state)
+
+        confirmed_order_lines = self.event_0.sale_order_lines_ids.filtered(lambda line: line.state == 'sale')
+        expected_price = sum(confirmed_order_lines.mapped('price_subtotal'))
+        self.assertEqual(self.event_0.sale_price_subtotal, expected_price)
```

Comment:
To check with the reviewer (⚠️ Do not change that. It's just a suggestion). I think we could maybe simplify the test even more if we discard all existing sale orders that have been created in the test set up methods and if we "hardcode" the expected price subtotal like so:

(untested)
```python
ticket1, ticket2 = self.event_0.event_ticket_ids
vals_list = [
    {'ticket': ticket1, 'price': 190.00, 'state': 'draft'},
    {'ticket': ticket2, 'price': 195.50, 'state': 'sale'},
    {'ticket': ticket2, 'price': 200.50, 'state': 'sent'},
    {'ticket': ticket1, 'price': 205.50, 'state': 'cancel'},
]
self.env['sale.order'].search([]).unlink() #  unlink all sales order
self.env['sale.order'].create([{
    'partner_id': self.env.user.partner_id.id,
    'state': vals['state'],
    'order_line': [
        (0, 0, {
            'event_id': self.event_0.id,
            'event_ticket_id': vals['ticket'].id,
            'product_id': vals['ticket'].product_id.id,
            'price_unit': vals['price'],
        })
    ]
} for vals in vals_list])
self.assertEqual(self.event_0.sale_price_subtotal, 195.50)
```

What I like about this approach is that we have all sale orders at glance. If we ever add a new sale order in the test setup methods, the test won't be impacted by the change as we delete all of them. We also avoid writing code to recompute the price subtotal in the assert.

What do you think?